lib/android/index.js

```diff
@@ -67,6 +67,78 @@ class Android {
 
     return this._downloadFile(sourcePath, destinationPath);
   }
+
+  async pidof(packageName) {
+    const cmd1 = `pidof ${packageName}`;
+    log.info(`pidof ${cmd1}`);
+    const proc1 = await this._runCommand(cmd1);
+    const ps1 = await adb.util.readAll(proc1);
+    var ps = ps1.toString();
+    log.info(`pidof ${ps}`);
+
+    if (ps && !Number.isNaN(Number.parseInt(ps.trim()))) {
+      return Number.parseInt(ps.trim());
+    }
+
+    const cmd2 = `ps -o PID,NAME | grep ' ${packageName}$'`;
+    log.info(`pidof ${cmd2}`);
+    const proc2 = await this._runCommand(cmd2);
+    const ps2 = await adb.util.readAll(proc2);
+    var ps = ps2.toString();
+    log.info(`pidof ${ps}`);
+
+    for (const line in ps.split(/[\r\n]+/)) {
+      const parts = line.trim().split(' ');
+      if (parts.length == 2 && parts[0] === packageName && !Number.isNaN(Number.parseInt(parts[1].trim()))) {
+        return Number.parseInt(parts[1].trim());
+      }
+    }
+
+    return null;
+  }
+
+  async _processStartTime(pid) {
+    // See https://stackoverflow.com/a/26538424 and others for
+    // determining process start time.  See
+    // https://stackoverflow.com/a/44524937 for determining USER_HZ
+    // (Linux kernel jiffies per second) without using syscall.  See
+    // https://stackoverflow.com/questions/16548528/command-to-get-time-in-milliseconds#comment33215571_16548827
+    // for the date formatting trick.
+    const MILLIS_PER_S = 1000;
+
+    const stat_and_timestamp = await this._runCommand(`date +%s%3N && awk '{print $1}' /proc/uptime && awk '{print $22}' /proc/self/stat /proc/${pid}/stat`);
```

Comment:
This line is a bit messy.  Let's put the command into it's own variable.

Reply:
Done.